src/models/FacebookGroup.ts

```diff
@@ -0,0 +1,9 @@
+export default interface FacebookGroup {
```

Comment:
Check out Partial<T>, it makes every field on a type optional.

This is the approach I took on the scraper model:

```
export type FacebookGroupAboutScrape = {
  name: string;
  description: string;
  isPublic: boolean;
  locations: string[];
  foundedOn: TimeStamp;
  memberCount: WholeNumber;
  memberCountIncreaseWeekly: WholeNumber;
  postCountIncreaseDaily: WholeNumber;
  postCountIncreaseMonthly: WholeNumber;
};

export type FacebookGroupAdminsScrape = {
  adminModeratorCount: WholeNumber;
  adminModeratorList: FacebookID[];
};

export type FacebookGroupPagesScrape = {
  pageCount: WholeNumber;
  pageList: FacebookID[];
};

export type FacebookGroupMembersScrape = {
  memberList: FacebookID[];
};

export type FacebookGroupScrape = {
  id: FacebookID;
  scrapedAt: TimeStamp;
  scrapeID: UUID;
} & FacebookGroupAboutScrape &
  FacebookGroupAdminsScrape &
  FacebookGroupPagesScrape &
  FacebookGroupMembersScrape;

// Examples of using Partial<T>
FacebookGroupScrape<FacebookGroupMembersScrape>
Partial<FacebookGroupMembersScrape>

type FacebookGroup = FacebookGroupAboutScrape &
  Partial<FacebookGroupAdminsScrape> &
  Partial<FacebookGroupPagesScrape> &
  Partial<FacebookGroupMembersScrape>;

// Moving towards more expressive types 
type Sample = FacebookGroupAboutScrape & {
  isPublic: true,
  someData: number
} | {
  isPublic: false,
   someData: never
}

Once we understand the edge cases we can refine our types
```